Plugin/AddTemplateVariable.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Mondu\Mondu\Plugin;
+
+use Mondu\Mondu\Helpers\Log as MonduLogger;
+use Mondu\Mondu\Helpers\Logger\Logger as MonduFileLogger;
+
+class AddTemplateVariable
+{
+    /**
+     * @var MonduLogger
+     */
+    private MonduLogger $monduLogger;
+
+    /**
+     * @var MonduFileLogger
+     */
+    private MonduFileLogger $monduFileLogger;
+
+    /**
+     * @param  MonduLogger      $monduLogger
+     * @param  MonduFileLogger  $monduFileLogger
+     */
+    public function __construct(
+        MonduLogger $monduLogger,
+        MonduFileLogger $monduFileLogger
+    ) {
+        $this->monduLogger = $monduLogger;
+        $this->monduFileLogger = $monduFileLogger;
+    }
+
+    /**
+     * @param  \Magento\Email\Model\Template  $subject
+     * @param  array                          $vars
+     *
+     * @return array[]
+     */
+    public function beforeSetVars(
+        \Magento\Email\Model\Template $subject,
+        array $vars
+    ) {
+        if (!$vars['order'] || !$vars['order']->getMonduReferenceId()) {
+            return [$vars];
+        }
+
+        try {
+            $monduReferenceId = $vars['order']->getMonduReferenceId();
+            $monduLog = $this->monduLogger->getLogCollection($monduReferenceId);
+
+            if (!$monduLog) {
+                return [$vars];
+            }
+
+            $vars['monduDetails'] = $this->prepareHtml($monduLog);
+        } catch (\Exception $e) {
+            $this->monduFileLogger->critical($e->getMessage());
+        }
+
+        return [$vars];
+    }
+
+    /**
+     * @param $monduLog
+     *
+     * @return string
+     */
+    protected function prepareHtml($monduLog)
+    {
+        $html = '';
+
+        if ($monduLog['payment_method']) {
+            $html .= '<strong>Payment Method:</strong> ' . ucwords(strtolower($monduLog['payment_method'])) . '<br />';
```

Comment:
**question:(blocking)** Why don't we provide each variable separately?

My initial concern was that the merchant would like to use localized text in their invoice. But maybe we can provide each variable separately and not worry about translations? WDYT?

Reply:
Thanks, @tikohov20, good question. I can wrap labels with the translation methods, it's ok. Regarding the different variables, instead of one variable, the merchant requested that. If it's a Mondu Payment - get everything with on variable HTML + Dynamic Data. If not - show an empty string. 
If it's ok to change the implementation and provide a couple of template variables instead of one - I'm ok with that and can quickly improve.

Reply:
Hmm, okey let's keep one variable for now. But we need to make sure that we have the strings translated for at least the merchants shop locale.

Reply:
Talked with Tigran and agreed that we will make a couple of translatable variables